src/Symfony/Component/HttpKernel/Event/GetResponseForExceptionEvent.php

```diff
@@ -19,45 +19,58 @@
  */
 class GetResponseForExceptionEvent extends RequestEvent
 {
-    /**
-     * The exception object.
-     *
-     * @var \Exception
-     */
-    private $exception;
+    private $throwable;
 
     /**
      * @var bool
      */
     private $allowCustomResponseCode = false;
 
-    public function __construct(HttpKernelInterface $kernel, Request $request, int $requestType, \Exception $e)
+    public function __construct(HttpKernelInterface $kernel, Request $request, int $requestType, $e)
     {
         parent::__construct($kernel, $request, $requestType);
 
-        $this->setException($e);
+        $this->throwable = $e;
     }
 
     /**
-     * Returns the thrown exception.
+     * Returns the thrown exception / error.
+     *
+     * @deprecated since Symfony 4.4, use "getThrowable()" instead
      *
-     * @return \Exception The thrown exception
+     * @return \Throwable The thrown exception / error
      */
     public function getException()
     {
-        return $this->exception;
+        @trigger_error(sprintf('The "%s::getException()" method is deprecated since Symfony 4.4, use "getThrowable()" instead.', \get_class($this)));
+
+        return $this->getThrowable();
+    }
+
+    public function getThrowable(): \Throwable
+    {
+        return $this->throwable;
     }
 
     /**
      * Replaces the thrown exception.
      *
      * This exception will be thrown if no response is set in the event.
      *
+     * @deprecated since Symfony 4.4, use "setThrowable()" instead
+     *
      * @param \Exception $exception The thrown exception
      */
     public function setException(\Exception $exception)
     {
-        $this->exception = $exception;
+        @trigger_error(sprintf('The "%s::setException()" method is deprecated since Symfony 4.4, use "setThrowable()" instead.', \get_class($this)));
+
+        $this->setThrowable($exception);
+    }
+
+    public function setThrowable(\Throwable $throwable): void
```

Comment:
I think you only added the new method because the name is fitting a little better or? Does not seem worth to me as the event class is still `ExceptionEvent`, so a `setException(\Throwable $throwable)` seems good enough to me.

Reply:
It's not about the name but about being able to have this in 4.4 without breaking changes. `setException(\Throwable $exception)` would break any classes extending this one with `setException(\Exception $exception)`

Reply:
I think Event classes should be marked as final. There is no point in people extending them as the libraries that use the event, will only dispatch this event. So extending events in user-land achieves nothing as the subclasses won't be dispatched.

Reply:
The event classes are final now and I think we can just change the types for this in master.

Reply:
Should be good now.